### ISSUE TITLE
To extract `<Trans/>` strings the scanner should use the same lib to parse html as react-i18next

### ISSUE THREAD

**Issue by messerm** [i18next/i18next-scanner#72]
Under certain circumstances, the order for the indexed tags are not the same for the parsed strings within `react-i18next` and `i18next-scanner`. Translation will fallback to key, if no trans key is specified and the generated key from the string is used.

Following text:
```
<Trans><p>Lorem Ipsum is simply dummy text of the printing and typesetting industry.</p>
		<p>Lorem Ipsum has been the industry's standard dummy text ever since the 1500s</p>
                </Trans>
```
`i18next-scanner@2.4.4` generated key:
`<0>Lorem Ipsum is simply dummy text of the printing and typesetting industry.</0> <2>Lorem Ipsum has been the industry's standard dummy text ever since the 1500s</2>`

`react-i18next@7.3.1` generated key:
`<0>Lorem Ipsum is simply dummy text of the printing and typesetting industry.</0><1>Lorem Ipsum has been the industry's standard dummy text ever since the 1500s</1>`

#### Version
- i18next: <i>10.2.2</i>
- i18next-scanner: <i>2.4.4</i>

#### Configuration
default konfiguration supporting `<trans/>`.

**Comment by cheton** [comment]
```jsx
<Trans>
    <p>Lorem Ipsum is simply dummy text of the printing and typesetting industry.</p>
    Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    <p>Lorem Ipsum has been the industrys standard dummy text ever since the 1500s</p>
</Trans>
```

will generate translation key as below:

```
<0>Lorem Ipsum is simply dummy text of the printing and typesetting industry.</0>Lorem Ipsum is simply dummy text of the printing and typesetting industry.<2>Lorem Ipsum has been the industrys standard dummy text ever since the 1500s</2>
```

---

```jsx
<Trans>
    Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    <p>
        Lorem Ipsum is simply dummy text of the printing and typesetting industry.
        <p>
            Lorem Ipsum is simply dummy text of the printing and typesetting industry.
        </p>
    </p>
    <p>{'Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s'}</p>
</Trans>
```

will generate translation key as below:

```js
Lorem Ipsum is simply dummy text of the printing and typesetting industry.<1>Lorem Ipsum is simply dummy text of the printing and typesetting industry.<1>Lorem Ipsum is simply dummy text of the printing and typesetting industry.</1></1><2>Lorem Ipsum has been the industry's standard dummy text ever since the 1500s</2> 
```

---

```jsx
<Trans>
    Click <Anchor>Me</Anchor> to continue.
</Trans>
```

will generate translation key as below:

```
Click <1>Me</1> to continue.
```

---

```jsx
<Trans>
    <div>Click <Anchor>Me</Anchor> to continue.</div>
</Trans>
```

will generate translation key as below:

```
<0>Click <1>Me</1> to continue.</0> 
```

**Comment by cheton** [comment]
There is an alternative JSX parser: https://github.com/RReverser/acorn-jsx

I will check if it can resolve the issue of handling whitespace characters in JSX.

**Comment by cheton** [comment]
Fixed in [v2.6.0](https://github.com/i18next/i18next-scanner/releases/tag/v2.6.0)

**Comment by kibaHanzo** [comment]
Facing this issue with version: 4.1.0

React code
`
          <Trans i18nKey={'test-key'}>
            <div>
              {'string1 '}
              {/*eslint-disable-next-line jsx-a11y/anchor-is-valid*/}
              <a>{'string2'}</a>
              {' string3'}
            </div>
          </Trans>
`

Expected translation to be generated (with i18next@19.1.0): 
`"test-key": "<0>string1 <1>string2</1> string3</0>" `

Translation getting generated:
`"test-key": "<0>string1 <2>string2</2> string3</0>" `
